commit b3acc68cafc4c996f7e1e00176b60ee1041769c4
Author: Phil Elwell <phil@raspberrypi.org>
Date:   Tue Nov 27 16:56:50 2018 +0000

    BCM270X_DT: Add new Ethernet DT parameters
    
    Add "eth_downshift_after" DT parameter to allow the delay before the
    downshift to be specified. The default is 2 auto-negotiation cycles,
    and legal values are 2, 3, 4, 5 and 0 (disabled).
    
    Add "eth_max_speed" DT parameter as a way of prohibiting 1000Mbps
    links. This can be used to avoid the delay until the downshift mechanism
    activates. Legal values are 10, 100 and 1000, where the default is
    unlimited (effectively 1000Mbps).
    
    Signed-off-by: Phil Elwell <phil@raspberrypi.org>

diff --git a/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi b/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi
index 69d2c2757e45..78742d623b6c 100644
--- a/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi
+++ b/arch/arm/boot/dts/bcm283x-rpi-lan7515.dtsi
@@ -50,5 +50,7 @@
 		tx_lpi_timer = <&eth_phy>,"microchip,tx-lpi-timer:0";
 		eth_led0 = <&eth_phy>,"microchip,led-modes:0";
 		eth_led1 = <&eth_phy>,"microchip,led-modes:4";
+		eth_downshift_after = <&eth_phy>,"microchip,downshift-after";
+		eth_max_speed = <&eth_phy>,"max-speed:0";
 	};
 };
diff --git a/arch/arm/boot/dts/overlays/README b/arch/arm/boot/dts/overlays/README
index fb6d0ddbbc76..1e89b61a4906 100644
--- a/arch/arm/boot/dts/overlays/README
+++ b/arch/arm/boot/dts/overlays/README
@@ -98,6 +98,11 @@ Params:
                                 compatible devices (default "on"). See also
                                 "tx_lpi_timer".
 
+        eth_downshift_after     Set the number of auto-negotiation failures
+                                after which the 1000Mbps modes are disabled.
+                                Legal values are 2, 3, 4, 5 and 0, where
+                                0 means never downshift (default 2).
+
         eth_led0                Set mode of LED0 (usually orange) (default
                                 "1"). The legal values are:
                                 0=link/activity          1=link1000/activity
@@ -108,6 +113,10 @@ Params:
         eth_led1                Set mode of LED1 (usually green) (default
                                 "6"). See eth_led0 for legal values.
 
+        eth_max_speed           Set the maximum speed a link is allowed
+                                to negotiate. Legal values are 10, 100 and
+                                1000 (default 1000).
+
         i2c_arm                 Set to "on" to enable the ARM's i2c interface
                                 (default "off")
 
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1691_b3acc68cafc4c996f7e1e00176b60ee1041769c4.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1691_b3acc68cafc4c996f7e1e00176b60ee1041769c4.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
